internal/k8s/gatewayclient/gatewayclient.go

```diff
@@ -388,15 +389,19 @@ func (g *gatewayClient) Update(ctx context.Context, obj client.Object) error {
 	return nil
 }
 
-func (g *gatewayClient) CreateOrUpdateDeployment(ctx context.Context, deployment *apps.Deployment, mutators ...func() error) (bool, error) {
-	operation, err := controllerutil.CreateOrUpdate(ctx, g.Client, deployment, func() error {
+func multiMutatorFn(mutators []func() error) func() error {
```

Comment:
abstracting this out into a function wrapper since we needed the same logic in one more place now for `CreateOrUpdateSecret`